File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
         def dfs(root,maxVal):
            if not root: return 0
            res=1 if(root.val>=maxVal) else 0
            res+=dfs(root.left,max(maxVal,root.val))
            res+=dfs(root.right,max(maxVal,root.val))
            return res
         return dfs(root,root.val)
        

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.java

class Solution {
    public int goodNodes(TreeNode root) {
        return dfs(root,-10000);
    }
    public int dfs(TreeNode root,int maxVal)
    {
        if(root==null) return 0;
        int res=(root.val>=maxVal)?1:0;
        res+=dfs(root.left,Math.max(maxVal,root.val));
        res+=dfs(root.right,Math.max(maxVal,root.val));
        return res;
    }
}